File: Week - 2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int main(void)
{
//Getting input from the user
    string text = get_string("Text: ");
    int letters = 0;
    int words = 1;
    int sentences = 0;

    for (int i = 0; i < strlen(text); i++)
    {
//Counts the number of letters
        if (isalpha(text[i]))
        {
            letters++;
        }
//Counts the number of words
        else if (text[i] == ' ')
        {
            words++;
        }
//Counts the number of sentences
        else if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            sentences++;
        }
    }
//Calculates the Grade using the Coleman-Liau index
    float L = letters / (float) words * 100;
    float S = sentences / (float) words * 100;
    int index = round(0.0588 * L - 0.296 * S - 15.8);

//The different outputs for different indices
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

File: Week - 6/sentimental-readability/readability.py
from cs50 import get_string
import math

# Get the text from the user
text = get_string("Text: ")
letters = float(0)
words = float(1)
sentences = float(0)

# Converting text to integer
n = len(text)

# counting the number of letters
for i in range(n):
    if (text[i] >= 'a' and text[i] <= 'z') or (text[i] >= 'A' and text[i] <= 'Z'):
        letters += 1

# counting the number of words
for i in range(n):
    if text[i].isspace():
        words += 1

# counting the number of sentences
for i in range(n):
    if text[i] == "." or text[i] == "!" or text[i] == "?":
        sentences += 1
        
# Calculating the index of grade
l = 100 * letters/words
s = sentences/words * 100
index = round(0.0588 * l - 0.296 * s - 15.8)
if index >= 16:
    print("Grade 16+")
elif index < 1:
    print("Before Grade 1")
else:
    print(f"Grade {index}")
